amqp-client/src/test/java/zipkin2/reporter/amqp/RabbitMQSenderRule.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2016-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.reporter.amqp;
+
+import java.time.Duration;
+import org.junit.AssumptionViolatedException;
+import org.junit.ClassRule;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import zipkin2.CheckResult;
+
+import static zipkin2.Call.propagateIfFatal;
+
+/** This should be used as a {@link ClassRule} as it takes a very long time to start-up. */
+class RabbitMQSenderRule extends ExternalResource {
+  static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQSenderRule.class);
+  static final String IMAGE = "rabbitmq:3.7-management-alpine";
```

Comment:
this image has rabbitmqadmin which is a lot more reliable than the java variants in my experience trying to get anything to pass without lots of sleeps.